src/Client/UnixSSLStream.hpp

```diff
@@ -366,6 +366,10 @@ int UnixSSLStream::connect(const ConnectOptions &opts_arg)
 
 	SSL_set_connect_state(ssl);
 
+	/* Trigger client-server negotiation. */
+	size_t rcvd;
+	SSL_read_ex(ssl, nullptr, 0, &rcvd);
```

Comment:
AFAIU to initiate a connection we can use `SSL_connect`, which would probably look better.

Anyway, I don't think that a stream needs to initiate a connection before it's read from or written to - we shouldn't add the stream fd to the event machine before we try to read from or write to it.

Reply:
I agree, after some iterations of refactoring there will be no polling before trying to read/write. But now it's too complicated.